### ISSUE TITLE
euler(3, inf) regression: nan instead of float inf

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#17612]
Before #16727, we had:
```py
>>> euler(3, float('inf'))
inf
```
Now (after #16727) we have 
```py
>>> euler(3, float('inf'))
nan
```
(note those are `float` inputs not `Float`).

Do we care?  Perhaps its not really a design goal but it is handy that SymPy functions generally behave like double-precision `import math` functions (e.g., lambdify falls back on them when there are no `math` function).  This generally happens because of mpmath.  When we sidestep mpmath, we sometimes lose that behaviour.

**Comment by oscarbenjamin** [comment]
I don't quite understand what you mean. Does mpmath have this function? Would a math-module style function return inf rather than nan here?

**Comment by cbm755** [comment]
I think SymPy uses mpmath's euler function for evaluation (haven't specifically checked but that is the usual way our special functions work).

> Would a math-module style function return inf rather than nan here?

Yes!  Re: math-module: usually (and ideally) double-precision functions give a consistent behaviour for infinity.  Like `math.exp(float('inf'))` gives `inf` rather than `nan` and `math.exp(-float('inf'))` gives `0.0`.

**Comment by cbm755** [comment]
Contrast this with `math.sin(float('inf'))` which gives nan.

**Comment by cbm755** [comment]
Correction: we call `mpmath.eulerpoly` for numerical evaluation.

**Comment by oscarbenjamin** [comment]
Did you mean to close this?

I think this is something that should be fixed where possible. We have this:
```julia
In [5]: exp(oo)                                                                                                                   
Out[5]: ∞

In [6]: exp(float('inf'))                                                                                                         
Out[6]: ∞
```
so either that should be changed or other functions like `euler` should be changed to match that behaviour. I think that giving `oo` when that is a well defined answer is desirable.

**Comment by cbm755** [comment]
Yes I did mean to close it, but I could retitle it instead as its not specific to `euler` (and isn't really a regression).
